### PR TITLE
Rewrite reviewing pull requests guide

### PR DESCRIPTION
**Pull request by eternoendless**
I have rewritten the guide for reviewing and approving pull requests in the maintainer's guide.

Here's what it looks like rendered:

Page preview

### CODE REVIEW

File: content/maintainers-guide/reviewing-pull-requests.md [hunk 1]
```diff
@@ -1,156 +1,215 @@
...
+|:-----------------------------------------------|:-------------------------------------------------------------|
+| <label type="type">Bug fix</label>             | This PR fixes a bug                                          |
+| <label type="type">Feature</label>             | This PR introduces a new feature                             |
+| <label type="type">Improvement</label>         | This PR is an improvement (e.g. performance)                 |
+| <label type="type">Refactoring</label>         | This PR is a refactoring                                     |
+| <label type="branch">1.7.8.x</label>           | Destination branch (e.g. 1.7.8.x, develop, ...)              |
+| <label type="bcbreak">BC break</label>         | This PR includes breaking changes                            |
+| <label type="keyfeature">Key feature</label>   | This PR includes a feature to be highlighted in this release |
+| <label type="wip">WIP</label>                  | This PR is a work in progress                                |
+| <label type="e2e">E2E</label>                  | This PR is about end-to-end automated tests                  |
```
Comment (boubkerbribri) [inline content/maintainers-guide/reviewing-pull-requests.md:198]: I think there is no such label anymore, it's `TE` and `E2E Tests`

Comment (eternoendless) [reply content/maintainers-guide/reviewing-pull-requests.md:198]: I see some open PRs with that label: https://github.com/PrestaShop/PrestaShop/pulls?q=is%3Aopen+is%3Apr+label%3A%22E2E+Tests%22

Question: is the "TE" label really needed?

Comment (matks) [reply content/maintainers-guide/reviewing-pull-requests.md:198]: E2E might be an old label

File: content/maintainers-guide/reviewing-pull-requests.md [hunk 2]
```diff
@@ -1,156 +1,215 @@
...
+Labeling helps project members keep track of the status of Pull Requests.
+
+Two bots monitor the project's Issues and Pull Requests on GitHub to help automate common tasks.
+
+- [Prestonbot][prestonbot] will evaluate whether there are missing/invalid items in Pull Requests' descriptions and label them accordingly.
+- [Issuebot][issuebot] helps sync the status of Issues with its linked Pull Request by applying labels and performing kaban board automation.
+
+### Action required
+
+The following labels are added when an action is required:
```
Comment (matks) [inline content/maintainers-guide/reviewing-pull-requests.md:172]: What about the label `Waiting for dev`?

**Review by matks — COMMENTED** [review]
Overall OK for me. I provided 1 question.

File: content/maintainers-guide/reviewing-pull-requests.md (hunk 1 above)
Comment (boubkerbribri) [reply content/maintainers-guide/reviewing-pull-requests.md:198]: > 

- `E2E TESTS` is the one used for UI tests.
- `TE` is a pull request category for all tests. and we used it to run lint on UI tests (we should use `E2E Tests` for that)
- `E2E` (without `TESTS`) is something i never used personally and should be deleted.

File: content/maintainers-guide/reviewing-pull-requests.md (hunk 2 above)
Comment (eternoendless) [reply content/maintainers-guide/reviewing-pull-requests.md:172]: In which case you would need it in a PR?

File: content/maintainers-guide/reviewing-pull-requests.md (hunk 1 above)
Comment (boubkerbribri) [reply content/maintainers-guide/reviewing-pull-requests.md]: `TE` is linked to the PR category (refacto, tests ...), if a PR has only modification on unit or integrations tests, it's not `E2E Tests` but the PR category is `TE`.

File: content/maintainers-guide/reviewing-pull-requests.md (hunk 2 above)
Comment (matks) [reply content/maintainers-guide/reviewing-pull-requests.md:172]: We use it when a developer action is needed, excluding review. For example if CI is failing for an unknown reason or specific expertise is required. I am not sure it's relevant to write it in this guide.

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @eternoendless

**Pull request merged**